crates/proof-of-sql/src/base/database/column_repetition_operation.rs

```diff
@@ -120,19 +136,48 @@ impl RepetitionOp for ColumnRepeatOp {
     fn op<T: Clone>(column: &[T], n: usize) -> impl Iterator<Item = T> {
         repeat_slice(column, n)
     }
+
+    fn op_fixed_size_binary(col_bytes: &[u8], width: usize, n: usize) -> Vec<u8> {
+        let num_rows = col_bytes.len() / width;
+        let mut out = Vec::with_capacity(col_bytes.len() * n);
+
+        for _ in 0..n {
+            for row_idx in 0..num_rows {
+                let start = row_idx * width;
+                let end = start + width;
+                out.extend_from_slice(&col_bytes[start..end]);
+            }
+        }
+        out
+    }
 }
 
 pub struct ElementwiseRepeatOp {}
 impl RepetitionOp for ElementwiseRepeatOp {
     fn op<T: Clone>(column: &[T], n: usize) -> impl Iterator<Item = T> {
         repeat_elementwise(column, n)
     }
+
+    fn op_fixed_size_binary(col_bytes: &[u8], width: usize, n: usize) -> Vec<u8> {
+        let num_rows = col_bytes.len() / width;
+        let mut out = Vec::with_capacity(col_bytes.len() * n);
+
+        for row_idx in 0..num_rows {
+            let start = row_idx * width;
+            let end = start + width;
+            let chunk = &col_bytes[start..end];
+            for _ in 0..n {
+                out.extend_from_slice(chunk);
+            }
+        }
+        out
+    }
```

Comment:
we can probably make this a little more idiomatic as well
```suggestion
fn op_fixed_size_binary(col_bytes: &[u8], width: usize, n: usize) -> Vec<u8> {
    let mut out = Vec::with_capacity(col_bytes.len() * n);
    out.extend(
        col_bytes
            .chunks_exact(width)
            .flat_map(|row| iter::repeat(row).take(n))
            .flatten()
            .cloned()
    );
    out
}
```